dbms/src/Storages/StorageDisaggregatedRemote.cpp

```diff
@@ -125,150 +130,178 @@ BlockInputStreams StorageDisaggregated::readFromWriteNode(
+                error.msg(),
+                batch_cop_task.store_addr,
+                retry_regions);
+
+            for (const auto & region : req->regions())
+            {
+                if (retry_regions.contains(region.region_id()))
                 {
-                    DB::DM::RemotePb::RemotePhysicalTable table;
-                    auto parse_ok = table.ParseFromString(physical_table);
-                    RUNTIME_CHECK_MSG(parse_ok, "failed to deserialize RemotePhysicalTable from response");
-
-                    Stopwatch watch_table;
-
-                    remote_tasks_from_wns[idx] = DM::RNRemoteTableReadTask::buildFrom(
-                        db_context,
-                        resp->store_id(),
-                        batch_cop_task.store_addr,
-                        snapshot_id,
-                        table,
-                        log);
-
-                    LOG_DEBUG(
-                        log,
-                        "Build RNRemoteTableReadTask finished, elapsed={:.3f}s store={} addr={} segments={} task_id={}",
-                        watch_table.elapsedSeconds(),
-                        resp->store_id(),
-                        batch_cop_task.store_addr,
-                        table.segments().size(),
-                        task_id);
+                    auto region_ver_id = pingcap::kv::RegionVerID(
+                        region.region_id(),
+                        region.region_epoch().conf_ver(),
+                        region.region_epoch().version());
+                    cluster->region_cache->dropRegion(region_ver_id);
                 }
-
-                this_elapse_ms = watch.elapsedMillisecondsFromLastTime();
-                summary.build_remote_task_ms += this_elapse_ms;
-                GET_METRIC(tiflash_disaggregated_breakdown_duration_seconds, type_build_task).Observe(this_elapse_ms / 1000.0);
-            });
+            }
+            throw Exception(
+                error.msg(),
+                ErrorCodes::DISAGG_ESTABLISH_RETRYABLE_ERROR);
+        }
+        else if (resp->error().has_error_locked())
+        {
+            using namespace pingcap;
+
+            const auto & error = resp->error().error_locked();
+
+            LOG_INFO(
+                log,
+                "Received EstablishDisaggregated response with retryable error: Key is locked, addr={} lock={}",
+                batch_cop_task.store_addr,
+                error.locked().DebugString());
+
+            // TODO: Better to construct the Backoff according to remaining backoff time.
+            kv::Backoffer bo(kv::copNextMaxBackoff);
+            kv::LockPtr lock = std::make_shared<kv::Lock>(error.locked());
+            std::vector<uint64_t> pushed;
+            std::vector<kv::LockPtr> locks{lock};
+            auto before_expired = cluster->lock_resolver->resolveLocks(bo, sender_target_mpp_task_id.query_id.start_ts, locks, pushed);
```

Comment:
Only retrying one LockException could greatly slow the query time when there are many unclean locks among all Regions. We need to change the logic in `LearnerRead` to return multiple LockException and resolve locks in a batch